### PR TITLE
[WIP] ConceptFusion

### PR DESCRIPTION
**Pull request by ykarmesh**
## Motivation and Context
Adds a concept fusion demo. Still a work in progress.

## How Has This Been Tested

Using the demo on a stored trajectory data.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist:

- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] I ran `tests/hw_manual_test.py` on hardware and observed that robot behavior looks normal.

### CODE REVIEW

File: projects/concept_fusion/concept_fusion.py [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+        Initialize concept fusion model.
+
+        Args:
+            args (DictConfig): Hydra config.
+        """
+        self.args = args
+
+        sam = sam_model_registry[args.model_type](checkpoint=Path(args.SAM_checkpoint_path))
+        sam.to(device=args.device)
+        self.mask_generator = SamAutomaticMaskGenerator(
```
[the remark this answers is not in the document]
Comment (ykarmesh) [reply projects/concept_fusion/concept_fusion.py:67]: Do you mean that someone will pass their own mask generator to ConceptFusion? Or do you mean that I just make all the hard-coded params part of the config?

File: projects/concept_fusion/concept_fusion.py [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+        bb_coords[i, :, 1] = torch.max(cluster, axis=0)[0]
+
+        # use different bright colors for points in different clusters
+        color_data[labels == i] = torch.tensor(COLOR_LIST[i % len(COLOR_LIST)]).float()
+    return bb_coords, color_data, labels
+
+class ConceptFusion:
+    def __init__(
+            self,
+            args: DictConfig,
```
[the remark this answers is not in the document]
Comment (ykarmesh) [reply projects/concept_fusion/concept_fusion.py:55]: It is a Hydra config.

File: projects/concept_fusion/concept_fusion.py [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+
+            xyz = torch.Tensor(self.camera.depth_to_xyz(depth.numpy())).reshape(-1, 3)[:, [0, 2, 1]]
+            xyz[:, 1] *= -1
+            xyz = (
+                torch.cat([xyz, torch.ones_like(xyz[..., [0]])], axis=1) @ camera_pose.T
+            )
+            self.voxel_map.add(
+                points=xyz[:, :3],
+                features=outfeat.reshape(-1, 1024),
+                rgb=torch.tensor(img).reshape(-1, 3),
```
[the remark this answers is not in the document]
Comment (ykarmesh) [reply projects/concept_fusion/concept_fusion.py:291]: This voxel class doesn't have the xyz_arg. That is a different voxel class
